convert2rhel/unit_tests/checks_test.py

```diff
@@ -283,13 +311,64 @@ def test_convert2rhel_latest_log_check_env(self, caplog, monkeypatch, convert2rh
-            [{"local_version": "0.25", "package_version": "convert2rhel-0:0.17-1.el7.noarch", "pmajor": "6"}],
-            [{"local_version": "0.25", "package_version": "convert2rhel-0:0.17-1.el7.noarch", "pmajor": "7"}],
-            [{"local_version": "0.25", "package_version": "convert2rhel-0:0.17-1.el7.noarch", "pmajor": "8"}],
-            [{"local_version": "1.10", "package_version": "convert2rhel-0:0.18-1.el7.noarch", "pmajor": "8"}],
+            [{"local_version": "0.17.0", "package_version": "convert2rhel-0:0.17.0-1.el7.noarch", "pmajor": "6"}],
+            [{"local_version": "0.17.0", "package_version": "Convert2RHEL Repository Info", "pmajor": "6"}],
+        ),
+        indirect=True,
+    )
+    def test_convert2rhel_latest_invalid_repoquery(self, caplog, convert2rhel_latest_version_test):
+        checks.check_convert2rhel_latest()
+
+        dummy_, package_version = convert2rhel_latest_version_test
+        log_msg = "Got a line without the C2R identifier: %s" % package_version
+        assert log_msg in caplog.text
+
+    @pytest.mark.skipif(pkgmanager.TYPE == "yum", reason="yum parsing function will return a different message")
+    @pytest.mark.parametrize(
+        ("convert2rhel_latest_version_test",),
+        (
+            [{"local_version": "0.17.0", "package_version": "C2R convert 2rhel-0:0.17.0-1.el7.noarch", "pmajor": "6"}],
+            [{"local_version": "0.17.0", "package_version": "C2R invalid_package.noarch", "pmajor": "6"}],
+        ),
+        indirect=True,
+    )
+    def test_convert2rhel_latest_value_error_dnf(self, caplog, convert2rhel_latest_version_test):
+        checks.check_convert2rhel_latest()
+
+        local_version, package_version = convert2rhel_latest_version_test
+        log_msg = (
+            "Invalid package string - %s, enter a package in one of the formats: NEVRA, NEVR, NVRA, NVR, ENVRA, ENVR."
+            % package_version[4:]
+        )
+        assert log_msg in caplog.text
+
+    @pytest.mark.skipif(pkgmanager.TYPE == "dnf", reason="dnf parsing function will return a different message")
+    @pytest.mark.parametrize(
+        ("convert2rhel_latest_version_test",),
+        (
+            [{"local_version": "0.17.0", "package_version": "C2R convert 2rhel-0:0.17.0-1.el7.noarch", "pmajor": "6"}],
+            [{"local_version": "0.17.0", "package_version": "C2R convert2rhel-0:0.1 7.0-1.el7.noarch", "pmajor": "6"}],
+        ),
+        indirect=True,
+    )
+    def test_convert2rhel_latest_value_error_yum(self, caplog, convert2rhel_latest_version_test):
+        checks.check_convert2rhel_latest()
+
+        local_version, package_version = convert2rhel_latest_version_test
+        # this is a test for a debug message, there is no other way to tell we had a ValueError in the parsing function
+        log_msg = "The following field(s) are invalid -"
+        assert log_msg in caplog.text
```

Comment:
Due to the comment about debug it seems like this test is asserting debug log messages. That is bad practice so we should not do it. Debug messages should not be tested unless they can spew out sensitive stuff we need to check for

Reply:
@SpyTec I added this test so we have coverage of the try/except blocking catching a ValueError that gets raised in the parsing function around line 166 in checks.py. Codecov will complain if I remove it, is that ok?

Reply:
We only care about the catch to log to debug anyway so yeah feel free. Code coverage is only a metric to see how we're doing, it is never a goal

Reply:
Understood, thanks! I'll go ahead and remove those tests